### PR TITLE
Feat/niloufar mont

### PR DESCRIPTION
**Pull request by lilmont**
A httpListenere that listens for requests and runs them through a pipleline with three pipes that routes urls semi-dynamically and handles exceptions and authentication.

**You can try these URLs like:**
http://localhost:7776/Products/GetAllProducts
http://localhost:7776/Products/GetUserById/{id}
http://localhost:7776/Orders/GetAllOrders
http://localhost:7776/Orders/GetOrderById/{id}

### CODE REVIEW

File: sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+public class AuthenticationPipe : BasePipe
+{
+    public AuthenticationPipe(Action<HttpContext> next) : base(next)
+    {
+    }
+
+    public override void Handle(HttpContext context)
+    {        
+        context.Response.OutputStream.Write(
```
Comment (thisisnabi) [inline sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs:13]: Why do you write on the output?

File: sources/Niloo Mont/MicroWebFramework/EncodingService.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Text;
+
+namespace MicroWebFramework;
+
+public static class EncodingService
+{
+    public static byte[] GetBytes(string input) => 
+        Encoding.UTF8.GetBytes(input+"\n");
```
Comment (thisisnabi) [inline sources/Niloo Mont/MicroWebFramework/EncodingService.cs:8]: don't abuse the content.

 remove `\n`!

File: sources/Niloo Mont/README.md
Comment (thisisnabi) [inline sources/Niloo Mont/README.md:1]: Thanks, it's good.

**Review by thisisnabi — CHANGES_REQUESTED** [review]

**Comment by thisisnabi** [comment]
Please do not choose 2 reviewers! @lilmont

File: sources/Niloo Mont/MicroWebFramework/EncodingService.cs (hunk 2 above)
Comment (lilmont) [reply sources/Niloo Mont/MicroWebFramework/EncodingService.cs:8]: Sure. How can I make the content more readable?

File: sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs (hunk 1 above)
Comment (lilmont) [reply sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs:13]: Where should I write?

File: sources/Niloo Mont/README.md
Comment (lilmont) [reply sources/Niloo Mont/README.md:1]: Thanks for the feedback <3

File: sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs (hunk 1 above)
Comment (thisisnabi) [reply sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs:13]: It's  log Nilofar! just write your action result into the response.

serialize action response and write into the response.

File: sources/Niloo Mont/MicroWebFramework/ProductsController.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+        new User() { Id = 4, Name = "User4" },
+        new User() { Id = 5, Name = "User5" },
+    };
+
+    // Products/GetUserById/{id}
+    public void GetUserById(int id)
+    {
+        if (!users.Any(p => p.Id == id))
+        {
+            _httpContext.Response.OutputStream.Write(
```
Comment (thisisnabi) [inline sources/Niloo Mont/MicroWebFramework/ProductsController.cs:32]: You can move the writing into the response to another place. maybe `EndpointPipe`

In The ASP.NET, It handle in the ResultFilters :)

The advantage of this is that you have performed an action globally.

File: sources/Niloo Mont/MicroWebFramework/EncodingService.cs (hunk 2 above)
Comment (thisisnabi) [reply sources/Niloo Mont/MicroWebFramework/EncodingService.cs:8]: You shouldn't have to do extra work in writing (serializing )the response, API developer is responsible for providing readable output. Currently you are SDK developer!

File: sources/Niloo Mont/README.md
Comment (thisisnabi) [reply sources/Niloo Mont/README.md:1]: I did not catch.
do you need more feedback? 

I look for the correct approach in the review.
The principle to choosing change for me is `So what?`.

File: sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs (hunk 1 above)
Comment (lilmont) [reply sources/Niloo Mont/MicroWebFramework/AuthenticationPipe.cs:13]: You're right. Thanks. I fixed it.

File: sources/Niloo Mont/README.md
Comment (lilmont) [reply sources/Niloo Mont/README.md:1]: No, I was just thanking :). This (<3) is the sign for the heart sticker.

**Review by thisisnabi — APPROVED** [review]

**Pull request merged**